### ISSUE TITLE
Show progress during db commits

### ISSUE THREAD

**Issue by andrewtoth** [casey/ord#1574]
Every 5k blocks the db is flushed to disk. This can be a long time and users just think the indexing has stopped. Showing some type of progress in the status bar could help with this.

**Comment by casey** [comment]
I'd love to show a progress bar. I'm not sure if the delay is on our side, or on the redb side. If it's the latter, this might require support from redb.

**Comment by andrewtoth** [comment]
Can't we have progress updated on its own thread? It would just listen on a channel for height updates or to show commit progress and show `.`, `..`, `...`, repeating while its committing? So users don't think its frozen?

**Comment by casey** [comment]
Redb is ACID, so commits are all or nothing, and partial commits will never be visible, so if the issue is long redb commits, we can't get height updates. Moving the progress bar into its own thread would work, but the progress bar displays progress based on block height. Adding a spinner to it would be good. Does indicatif support this? If so, adding a spinner might be a one-line change.

**Comment by casey** [comment]
I made a [great spinner for imdl, a bittorrent utility](https://github.com/casey/intermodal/blob/2a748fc5338db865d634578f95354db792f19d96/src/consts.rs#L45) so we could reuse the characters from that, which count from 0 to 255 in binary using braille.

**Comment by andrewtoth** [comment]
Right, I meant spinner. Sorry for the confusion. I understand we can't show progress of how far along, but we can show to the user that the application is not frozen by updating a spinner.

**Comment by jpistone** [comment]
> Right, I meant spinner. Sorry for the confusion. I understand we can't show progress of how far along, but we can show to the user that the application is not frozen by updating a spinner.

Ideally with an explanatory comment along the lines of "Committing batch, this may take up to 5 minutes..."

Otherwise to the user it may just look frozen _with style_.
